### PR TITLE
Separate calls migration

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

I've migrated the calls/forms to use `submit` and `source`.

I've also handled changes from https://github.com/mattermost/mattermost-plugin-apps/pull/298, e.g. the case of `context.team.id` instead of `context.team_id`.

### CODE REVIEW

**Comment by mickmister** [comment]
@levb I believe this is fully migrated for Apps 1.0 now

File: src/types/apps.ts [hunk 1]
```diff
@@ -2,9 +2,249 @@
 import {UserProfile} from 'mattermost-redux/types/users';
 import {Post} from 'mattermost-redux/types/posts';
 import {Channel} from 'mattermost-redux/types/channels';
-import {AppCall, AppCallValues, AppContext} from 'mattermost-redux/types/apps';
+import {Team} from 'mattermost-redux/types/teams';
 
-import {StoredOauthUserToken} from 'utils/ZDTypes';
+import {StoredOauthUserToken} from 'types/zendesk';
```
[the remark this answers is not in the document]
Comment (mickmister) [reply src/types/apps.ts:7]: @jfrerich A couple things about this:

- This specific type may be different across Apps. But in that case, they can just use their own type.
- We are in the process of creating an npm package `@mattermost/apps` that will contain lots of these kinds of types, and decouple them from the current ones available, that are mainly curated for the frontend's usage.

File: src/utils/helper_classes/fields/fields_builder.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import {AppCallRequest, AppCallValues, AppField, AppFormValue} from 'mattermost-redux/types/apps';
-import {AppFieldTypes} from 'mattermost-redux/constants/apps';
+import {AppFieldTypes} from '../../../constants/apps';
```
[the remark this answers is not in the document]
Comment (mickmister) [reply src/utils/helper_classes/fields/fields_builder.ts:1]: @jfrerich I wonder what project you've been working on that has this convention 🙂 Sounds like a good HW ticket

[the remark this answers is not in the document]
Comment (mickmister) [reply src/utils/helper_classes/fields/fields_builder.ts:1]: HW ticket here https://github.com/mattermost/mattermost-app-zendesk/issues/100

File: src/types/apps.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfrerich) [reply src/types/apps.ts:7]: Sounds great! Thanks for the update

**Comment by mickmister** [comment]
Closing since https://github.com/mattermost/mattermost-app-zendesk/pull/95 contains these changes

**Pull request closed without merging**